marklogic-data-hub/src/main/java/com/marklogic/hub/deploy/commands/LoadUserArtifactsCommand.java

```diff
@@ -188,44 +201,45 @@ public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) th
                     }
                 });
             }
-
-            // let's do step-definitions
-            if (stepDefPath.toFile().exists()) {
-                Files.walkFileTree(stepDefPath, new SimpleFileVisitor<Path>() {
-                    @Override
-                    public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
-                        executeWalk(
-                            dir,
-                            stepDefModulesFinder,
-                            propertiesModuleManager,
-                            stepResourceToURI,
-                            buildMetadata(hubConfig.getStepDefinitionPermissions(), "http://marklogic.com/data-hub/step-definition"),
-                            stagingStepDefDocumentWriteSet,
-                            finalStepDefDocumentWriteSet
-                        );
-                        return FileVisitResult.CONTINUE;
-                    }
-                });
-            }
-
-
-            // let's do flows
-            if (flowPath.toFile().exists()) {
-                Files.walkFileTree(flowPath, new SimpleFileVisitor<Path>() {
-                    @Override
-                    public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
-                        executeWalk(
-                            dir,
-                            flowDefModulesFinder,
-                            propertiesModuleManager,
-                            flowResourceToURI,
-                            buildMetadata(hubConfig.getFlowPermissions(), "http://marklogic.com/data-hub/flow"),
-                            stagingFlowDocumentWriteSet,
-                            finalFlowDocumentWriteSet
-                        );
-                        return FileVisitResult.CONTINUE;
-                    }
-                });
+            ArtifactManager artifactManager = new ArtifactManagerImpl(hubConfig);
+            List<String> roles = new LinkedList<>();
```

Comment:
```suggestion
            Set<String> roles = new HashSet<>();
```

Reply:
Refactored to add a canUserUpdate property artifactTypeInfo class, so checking the roles is no longer necessary.